pkg/aws/ec2/ec2.go

```diff
@@ -118,48 +118,32 @@ func deriveStatus(err error) string {
 
 // describeNetworkInterfaces lists all ENIs
 func (c *Client) describeNetworkInterfaces(ctx context.Context, subnets ipamTypes.SubnetMap) ([]ec2_types.NetworkInterface, error) {
-	var (
-		networkInterfaces []ec2_types.NetworkInterface
-		interfacesFilters []ec2_types.Filter
-		nextToken         string
-	)
-
-	for {
-		c.limiter.Limit(ctx, "DescribeNetworkInterfaces")
-		input := &ec2.DescribeNetworkInterfacesInput{}
-		if nextToken != "" {
-			input.NextToken = &nextToken
+	result := []ec2_types.NetworkInterface{}
```

Comment:
small nit, the slice doesn't need to be initialized, can be `var result []ec2_types.NetworkInterface`

Reply:
Updated